File: sdkwork-backend-sdk-go/api/invocation_record.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/backend-sdk/types"
    sdkhttp "github.com/sdkwork/backend-sdk/http"
)

type InvocationRecordApi struct {
    client *sdkhttp.Client
}

func NewInvocationRecordApi(client *sdkhttp.Client) *InvocationRecordApi {
    return &InvocationRecordApi{client: client}
}

// Update an existing invocation record
func (a *InvocationRecordApi) Update(body sdktypes.PlusInvokeRecordForm) (sdktypes.PlusApiResultPlusInvokeRecordVO, error) {
    raw, err := a.client.Put(BackendApiPath("/record"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusInvokeRecordVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusInvokeRecordVO](raw)
}

// Create a new invocation record
func (a *InvocationRecordApi) Create(body sdktypes.PlusInvokeRecordForm) (sdktypes.PlusApiResultPlusInvokeRecordVO, error) {
    raw, err := a.client.Post(BackendApiPath("/record"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusInvokeRecordVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusInvokeRecordVO](raw)
}

// Get invocation records by page
func (a *InvocationRecordApi) ListByPage(body *sdktypes.QueryListForm, query map[string]interface{}) (sdktypes.PlusApiResultPagePlusInvokeRecordVO, error) {
    raw, err := a.client.Post(BackendApiPath("/record/list"), body, query, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPagePlusInvokeRecordVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPagePlusInvokeRecordVO](raw)
}

// Get all invocation records
func (a *InvocationRecordApi) ListAllEntities(body *sdktypes.QueryListForm) (sdktypes.PlusApiResultListPlusInvokeRecordVO, error) {
    raw, err := a.client.Post(BackendApiPath("/record/list/all"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusInvokeRecordVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusInvokeRecordVO](raw)
}

// Get an invocation record by ID
func (a *InvocationRecordApi) GetById(id string) (sdktypes.PlusApiResultPlusInvokeRecordVO, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/record/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultPlusInvokeRecordVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusInvokeRecordVO](raw)
}

// Delete an invocation record
func (a *InvocationRecordApi) Delete(id string) (sdktypes.PlusApiResultBoolean, error) {
    raw, err := a.client.Delete(BackendApiPath(fmt.Sprintf("/record/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultBoolean
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultBoolean](raw)
}

File: sdkwork-backend-sdk-python/sdkwork_backend_sdk/api/invocation_record.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import PlusApiResultBoolean, PlusApiResultListPlusInvokeRecordVO, PlusApiResultPagePlusInvokeRecordVO, PlusApiResultPlusInvokeRecordVO, PlusInvokeRecordForm, QueryListForm

class InvocationRecordApi:
    """invocation_record API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def update(self, body: PlusInvokeRecordForm) -> PlusApiResultPlusInvokeRecordVO:
        """Update an existing invocation record"""
        return self._client.put(f"/backend/v3/api/record", json=body)

    def create(self, body: PlusInvokeRecordForm) -> PlusApiResultPlusInvokeRecordVO:
        """Create a new invocation record"""
        return self._client.post(f"/backend/v3/api/record", json=body)

    def list_by_page(self, body: Optional[QueryListForm] = None, params: Optional[Dict[str, Any]] = None) -> PlusApiResultPagePlusInvokeRecordVO:
        """Get invocation records by page"""
        return self._client.post(f"/backend/v3/api/record/list", json=body, params=params)

    def list_all_entities(self, body: Optional[QueryListForm] = None) -> PlusApiResultListPlusInvokeRecordVO:
        """Get all invocation records"""
        return self._client.post(f"/backend/v3/api/record/list/all", json=body)

    def get_by_id(self, id: str) -> PlusApiResultPlusInvokeRecordVO:
        """Get an invocation record by ID"""
        return self._client.get(f"/backend/v3/api/record/{id}")

    def delete(self, id: str) -> PlusApiResultBoolean:
        """Delete an invocation record"""
        return self._client.delete(f"/backend/v3/api/record/{id}")
